servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHost.java

```diff
@@ -204,7 +218,14 @@ public boolean markExpired() {
 
     @Override
     public Single<C> newConnection(
-            Predicate<C> selector, final boolean forceNewConnectionAndReserve, @Nullable final ContextMap context) {
+            Predicate<C> selector, final boolean forceNewConnectionAndReserve, @Nullable ContextMap context) {
+        // We need to put our address latency tracker in the context for consumption.
+        if (healthIndicator != null) {
+            if (context == null) {
+                context = new DefaultContextMap();
+            }
+            context.put(REQUEST_TRACKER_KEY, healthIndicator);
```

Comment:
It's not ideal that we have to rely on context. Let's see how we retrieve it from the context in follow-up PRs, maybe there will be a better way.

Reply:
I'm open to suggestions. This is The best both Thomas and I have thought of so far playing within the constraints of `ConnectionFactory` and the `*Requester` interfaces.